src/main/java/ConsoleCommunicator.java

```diff
@@ -17,99 +17,113 @@ public void run() {
         }
     }
 
-    public String chooseGame() {
-        askGameMode();
-        String choice = takeHumanInput();
-        return gameLogic.findGameMode(choice);
-    }
-
-    public String takeHumanInput() {
-        return consoleDisplay.takeInput();
-    }
-
     public void runComputerGame() {
+        String statusMessage;
         displayGrid();
         while (gameOngoing()) {
             if (itIsHumanTurn()) {
-                runHumanTurn();
+                statusMessage = runHumanTurnReturnStatusMessage();
+                announceHumanSquareChoice();
             } else {
-                runComputerTurn();
-            }
-            if (gameIsWon()) {
-                announceWinner();
+                statusMessage = computerTurnStatus();
+                announceComputerTurn();
             }
             displayGrid();
-            gameLogic.alternatePlayer();
+            announceIfGameOver(statusMessage);
         }
     }
 
-    public void displayGrid() {
-        consoleDisplay.displayGrid(gameLogic.getGridSquares());
+    public void runHumanGame() {
+        String statusMessage;
+        displayGrid();
+        while (gameOngoing()) {
+            statusMessage = runHumanTurnReturnStatusMessage();
+            announceHumanSquareChoice();
+            displayGrid();
+            announceIfGameOver(statusMessage);
+        }
     }
 
-    public boolean gameOngoing() {
-        return gameLogic.gameOngoing();
+    public String runHumanTurnReturnStatusMessage() {
```

Comment:
The method name indicates that the method is violating one of the SOLID principles. Can you guess which one I'm thinking of? 😸